### DISCUSSION TITLE
Variable media Properties function with Z axis

### DISCUSSION THREAD

**Discussion by khashayarkhorsand** [ankrh/MCmatlab#24] [Q&A] [answered]
Dear Anders,

Firstly,I would like to thank you and other contributors starting from introducing  this wonderfull package codein 2018 onward. My question is about the possibility of implementing media Properties function such as scattering coeficient variable with Z axis in depth?shall I change the musfunc2(wavelength,FR,T,FD) to include the possibility of taking Z coortinate as variable also?Something like the convective boundary condition for temprature initial condition variabel linearly in  Z. 
Kind regards
Khashayar

**Comment by ankrh** [comment, marked as answer]
Thanks for your kind words!

The best way to do what you're asking for in MCmatlab would be to "bin" the tissue into different depths. In other words, you can split your medium into three media "low depth tissue", "medium depth tissue", "high depth tissue". You could even do this with a for loop that uses a large number of bins such as 100, although the geometry illustration will probably look terrible. The limit for the total number of (sub-)media is 256.

This approach is in principle the same as what MCmatlab does when simulating media that are dependent on fluence rate, temperature or fractional damage. Those media are also split into sub-media as far as the Monte Carlo solver is concerned.

- Anders

**Reply by khashayarkhorsand** [reply]
Thank you for your reply. The method by defining several media stacked in Z direction with variable scattering coefient was crossed my mind before but since my geometry is being imported as a chunk of STL and then voxelized ,it was not possible easily. I will divide up the whole geometry along Z axis with some basic geometry in a for loop and stack it along Z and then import the last complicated parts by STL.
Thank you again for your hint about the 256 limit of the total number of sub-medias.
Khashayar
